### PR TITLE
test(developer): keyboard info compiler unit tests

### PR DESCRIPTION
**Pull request by markcsinclair**
Tests for the keyboard info compiler

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by markcsinclair** [comment]
Coverage is improving over keyboard-info-compiler.ts, but more tests are needed for `fillLanguages()`, `fontSourceToKeyboardInfoFont()` and the latter part of `run()`.

File: developer/src/kmc-keyboard-info/test/test-keyboard-info-compiler.ts [hunk 1]
```diff
@@ -50,4 +111,342 @@ describe('keyboard-info-compiler', function () {
...
+  });
+  
+  it('check loadKmxFiles returns empty array if .kmx file is missing from .kmp', async function() {
+    const kpsFilename = KHMER_ANGKOR_KPS;
+    const compiler = new KeyboardInfoCompiler();
+    const kmpCompiler = new KmpCompiler();
+    assert.isTrue(await kmpCompiler.init(callbacks, {}));
+    const kmpJsonData = kmpCompiler.transformKpsToKmpObject(kpsFilename);
+    assert.isNotNull(kmpJsonData);
+    // remove .kps file
```
Comment (mcdurdin) [inline developer/src/kmc-keyboard-info/test/test-keyboard-info-compiler.ts:262]:
```suggestion
    // remove .kmx file
```

**Review by mcdurdin — APPROVED** [review]
Nits only, LGTM

File: developer/src/kmc-keyboard-info/test/test-keyboard-info-compiler.ts [hunk 2]
```diff
@@ -50,4 +111,342 @@ describe('keyboard-info-compiler', function () {
...
   });
+
+  it('check preinit creates langtagsByTag correctly', async function() {
+    const compiler = new KeyboardInfoCompiler(); // indirectly call preinit()
+    assert.isNotNull(compiler);
+    assert.deepEqual(langtags.find(({ tag }) => tag === 'en'), EN_LANGTAG);
+    assert.deepEqual((<any>unitTestEndpoints.langtagsByTag)['en'], EN_LANGTAG);
+    assert.deepEqual((<any>unitTestEndpoints.langtagsByTag)['en-Latn-US'], EN_LANGTAG);
+    assert.deepEqual((<any>unitTestEndpoints.langtagsByTag)['en-Latn'], EN_LANGTAG);
+    assert.deepEqual((<any>unitTestEndpoints.langtagsByTag)['en-US'], EN_LANGTAG);
```
[the remark this answers is not in the document]
Comment (markcsinclair) [reply developer/src/kmc-keyboard-info/test/test-keyboard-info-compiler.ts:122]: I have replaced the const tag with a dynamic one extracted from langtags ... should take less maintenance.

File: developer/src/kmc-keyboard-info/test/test-keyboard-info-compiler.ts (hunk 1 above)
Comment (markcsinclair) [reply developer/src/kmc-keyboard-info/test/test-keyboard-info-compiler.ts:262]: Good catch.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.300-beta](https://keyman.com/downloads/releases/beta/17.0.300)

**Pull request merged**